Content.Shared/Clumsy/ClumsySystem.cs

```diff
@@ -0,0 +1,129 @@
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeHyposprayInjectsEvent>(BeforeHyposprayEvent);
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeDefibrillatorZapsEvent>(BeforeDefibrillatorZapsEvent);
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeGunShotEvent>(BeforeGunShotEvent);
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeClimbEvent>(OnBeforeClimbEvent);
+    }
+
+    // If you add more clumsy interactions add them in this section!
+    #region Clumsy interaction events
+    private void BeforeHyposprayEvent(Entity<ClumsyComponent> ent, ref SelfBeforeHyposprayInjectsEvent args)
+    {
+        // Clumsy people sometimes inject themselves! Apparently syringes are clumsy proof...
+        if (!_random.Prob(ent.Comp.ClumsyDefaultCheck))
+            return;
+
+        args.TargetGettingInjected = args.EntityUsingHypospray;
+        args.InjectMessageOverride = "hypospray-component-inject-self-clumsy-message";
+        _audio.PlayPvs(ent.Comp.ClumsySound, ent);
+    }
+
+    private void BeforeDefibrillatorZapsEvent(Entity<ClumsyComponent> ent, ref SelfBeforeDefibrillatorZapsEvent args)
+    {
+        // Clumsy people sometimes defib themselves!
+        if (!_random.Prob(ent.Comp.ClumsyDefaultCheck))
+            return;
+
+        args.DefibTarget = args.EntityUsingDefib;
+        _audio.PlayPvs(ent.Comp.ClumsySound, ent);
+
+    }
+
+    private void BeforeGunShotEvent(Entity<ClumsyComponent> ent, ref SelfBeforeGunShotEvent args)
+    {
+        // Clumsy people sometimes can't shoot :(
+
+        if (args.Gun.Comp.ClumsyProof == true)
+            return;
+
+        if (!_random.Prob(ent.Comp.ClumsyDefaultCheck))
+            return;
+
+        _stun.TryParalyze(ent, ent.Comp.GunShootFailStunTime, true);
+
+        // Apply salt to the wound ("Honk!") (No idea what this comment means)
+        _audio.PlayPvs(ent.Comp.GunShootFailSound, ent);
+        _audio.PlayPvs(ent.Comp.ClumsySound, ent);
+
+        _popup.PopupEntity(Loc.GetString("gun-clumsy"), ent, ent);
+        args.Cancel();
+    }
+
+    private void OnBeforeClimbEvent(Entity<ClumsyComponent> ent, ref SelfBeforeClimbEvent args)
+    {
+        // This event is called in shared, thats why it has all the extra prediction stuff.
+        var rand = new System.Random((int)_timing.CurTick.Value);
+
+        // If someone is putting you on the table, always get past the guard.
+        if (!_cfg.GetCVar(CCVars.GameTableBonk) && args.PuttingOnTable == ent.Owner && !rand.Prob(ent.Comp.ClumsyDefaultCheck))
+            return;
+
+        HitHeadOnTableClumsy(ent, args.BeingClimbedOn);
+
+        _audio.PlayPredicted(ent.Comp.ClumsySound, ent, ent);
+
+        _audio.PlayPredicted(ent.Comp.TableBonkSound, ent, ent);
+
+        var beingClimbedName = Identity.Entity(args.BeingClimbedOn, EntityManager);
```

Comment:
Why do tables need an identity?

Reply:
From what I remember (After changing it and realizing the why I did it this way 😆) is because the Name function doesn't work on them for some reason! I'm not super good with the name functions so I could be missing something.

Reply:
But since the table does not have an IdentityComponent this just returns the table entity again
![grafik](https://github.com/user-attachments/assets/39d2c1a6-cb65-4fd8-91d6-abe587714e33)

Reply:
Ah I see! I'll take a look 🫡